Jenkinsfile

```diff
@@ -27,18 +28,8 @@ def generateParallelSteps(labels) {
         parallelNodes[label] = {
             node(label) {
                 if (isUnix()) {
-                    sh '''
-                        uname -a
-                        if test -e /etc/os-release; then
-                            cat /etc/os-release
-                        fi
-                        if test -e /proc/cpuinfo; then
-                            cat /proc/cpuinfo
-                        fi
-                        if test -e /proc/meminfo; then
-                            cat /proc/meminfo
-                        fi
-                    '''
+                    checkout scm
+                    sh 'bash ./checks.sh '+label
```

Comment:
if I remember correctly, the main concern is for the file rights (execution) and then the fact of potential nonexistence of bash as you mentioned.

Reply:
What @smerle33 is correct: `bash` is a requirement for the script (as per its shebang at least) so it's now explicit in the pipeline (Particularly since the `sh` pipeline instruction is using `sh` by default) call and the execution bit is not always guaranteed to be applied on the shell script (depends on how it is retrieved and on the underlying filesystem).